extension/js/common/helpers.ts

```diff
@@ -16,11 +16,18 @@ export const isFesUsed = async (acctEmail: string) => {
   return Boolean(fesUrl);
 };
 
-export const saveKeysAndPassPhrase = async (acctEmail: string, prvs: Key[], ppOptions?: PassphraseOptions) => {
+export const saveKeysAndPassPhrase = async (acctEmail: string, prvs: Key[], ppOptions?: PassphraseOptions, replaceKeys: boolean = false) => {
   const clientConfiguration = await ClientConfiguration.newInstance(acctEmail);
+  if (replaceKeys) {
+    await KeyStore.set(acctEmail, await Promise.all(prvs.map(KeyUtil.keyInfoObj))); // todo: duplicate identities
+    // todo: should we delete passphrases matching the deleted keys?
+  }
   for (const prv of prvs) {
-    await KeyStore.add(acctEmail, prv);
+    if (!replaceKeys) {
+      await KeyStore.add(acctEmail, prv);
+    }
     if (ppOptions !== undefined) {
+      // todo: perhaps it's easier just to store a set of passphrases without specifying longids?
```

Comment:
@tomholub 
I have a suspicion that we don't necessarily need to store passphrase connected to a particular key, since we already implemented the algo to search for keys that can be unlocked by a passphrase. We can simply store a set (array) of passphrases under `..._passphrases` key -- it would be easier to delete all the passphrases for the account.

Reply:
In principle, but if you have many keys (with many pass phrases) and you'd like to decrypt a key, you'll need to try all of them, which takes time (about 50ms per attempt last time I checked, but may have been improved since).

Reply:
That situation wouldn't happen for people who use EKM (we enforce same pass phrase for all), but could happen for other users.